File: Java/first_bad_version.java
/* The isBadVersion API is defined in the parent class VersionControl.
      boolean isBadVersion(int version); */

public class Solution extends VersionControl {
    public int firstBadVersion(int n) {
        int s = 1, e = n, ans = 0;
        while(s <= e){
            int m = s+(e-s)/2;
            if(isBadVersion(m)){
                ans = m;
                e = m - 1;
            }else{
                s = m + 1;
            }
        }
        return ans;
    }
}

File: CPP/Leet Code Solutions/first-bad-version/first-bad-version.cpp
bool isBadVersion(int version);
class Solution
{
public:
    int firstBadVersion(int n)
    {
        long long l = 1, r = n, mid;
        while (l <= r)
        {
            mid = (l + r) / 2;
            int checkbad = isBadVersion(mid);
            if (checkbad == true && isBadVersion(mid - 1) == false)
            {
                return mid;
            }
            else if (checkbad == false)
            {
                l = mid + 1;
            }
            else
            {
                r = mid;
            }
        }
        return 0;
    }
};